src/lib/components/MarkModal.svelte

```diff
@@ -17,24 +17,24 @@
   function saveFlags() {
     const selectedCardIds = Array.from($cardsSelectedStore);
 
-    localCards.map(card => {
-      if (selectedCardIds.includes(card.id)) {
-        switch (flag) {
-          case 'critical':
-            card.isCritical = !card.isCritical;
-            break;
-          case 'chop-move':
-            card.isChopMoved = !card.isChopMoved;
-            break;
-          case 'finesse':
-            card.isFinessed = !card.isFinessed;
-            break;
+    cards.updateCards((currentCards) => {
+      const updated = currentCards.map((card) => {
+        if (selectedCardIds.includes(card.id)) {
+          switch (flag) {
+            case 'critical':
+              card.isCritical = !card.isCritical;
+              break;
+            case 'chop-move':
+              card.isChopMoved = !card.isChopMoved;
+              break;
+            case 'finesse':
+              card.isFinessed = !card.isFinessed;
+              break;
+          }
         }
-      }
-      return card
-    });
-    cards.updateCards(localCards => {
-      return localCards;
+        return card;
+      });
+      return updated;
     });
```

Comment:
IIUC, this is the main bit that matters. The `updateCards` function takes a `Card[]` of the current card states and returns a new `Card[]` of the updated values. The problem is that you were taking the current card statuses under the name `localCards` and returning it straight away, so nothing is actually updated in the store.

Your `localCards.map` above wasn't actually doing anything (I think). Because the `map` method returns a new array, it doesn't modify in place. And even if it did, the binding in `updateCards` overwrites it for that scope.

Reply:
Okay, cool, thanks for explaining. Svelte stores are cool, but sometimes quite picky :/